### PR TITLE
SDL_CreateWindowWithPosition() and SDL_CreateWindowFrom() have been replaced with SDL_CreateWindowWithProperties()

### PR DESCRIPTION
**Pull request by slouken**
This is a more general interface that can be extended in the future as needed.

### CODE REVIEW

File: include/SDL3/SDL_render.h [hunk 1]
```diff
@@ -413,50 +479,94 @@ extern DECLSPEC SDL_Texture *SDLCALL SDL_CreateTexture(SDL_Renderer *renderer, U
...
  * The following properties are provided by SDL:
  *
  * With the direct3d11 renderer:
  *
- * ```
- * "SDL.texture.d3d11.texture" (pointer) - the ID3D11Texture2D associated with the texture
- * "SDL.texture.d3d11.texture_u" (pointer) - the ID3D11Texture2D associated with the U plane of a YUV texture
- * "SDL.texture.d3d11.texture_v" (pointer) - the ID3D11Texture2D associated with the V plane of a YUV texture
- * ```
+ * - "SDL.texture.d3d11.texture" (pointer) - the ID3D11Texture2D associated with the texture
```
Comment (madebr) [inline include/SDL3/SDL_render.h:546]: Does the documentation mention anywhere that properties are "read-only" (in the sense that a SDL user cannot override them)?
Or can they?

Comment (slouken) [reply include/SDL3/SDL_render.h:546]: They are, and that's a good idea, thanks.

**Comment by slouken** [comment]
@icculus, as a sanity check, does this seem like a good idea going forward?

**Comment by icculus** [comment]
I would be nervous about doing this to _every_ object in SDL, but this couples _way_ more closely than hints, and I like the hell out of that.

**Comment by slouken** [comment]
Yep, I'm just doing this for things where you want access to the underlying OS level object or potentially complex options that aren't expressible via simple function parameters. It seems to be working well so far.

Thanks!

**Pull request merged**